### ISSUE TITLE
[bug] Conan 2.0.11+ choose wrong default cppstd for apple-clang

### ISSUE THREAD

**Issue by sizeak** [conan-io/conan#14835]
### Environment details

* Operating System+version: macOS 13.6
* Compiler+version: apple-clang 14.0
* Conan version: 2.0.11 & 2.0.12
* Python version: 3.11.5

### Steps to reproduce

```
python -m pip install --upgrade conan==2.0.11
conan profile detect --force
```

I believe this is caused by the removal of 
```
    if compiler != "msvc":
        cppstd = _cppstd_default(compiler, version)
        if compiler == "apple-clang" and version >= "11":
            # forced auto-detection, gnu98 is too old
            cppstd = "gnu17"
        result.append(("compiler.cppstd", cppstd))
```
in this commit https://github.com/conan-io/conan/commit/7f6430ad621cc4b4c2871d47686bb4d9993de4fb#diff-054375ca45a4ad879799d84fd9d02d0aae4e517cf4a6be98134ff446712d833d

### Logs

```detect_api: Found apple-clang 14.0
detect_api: apple-clang>=13, using the major as version

Detected profile:
WARN: This profile is a guess of your environment, please check it.
WARN: Defaulted to cppstd='gnu17' for apple-clang.
WARN: The output of this command is not guaranteed to be stable and can change in future Conan versions.
WARN: Use your own profile files for stability.
Saving detected profile to /Users/runner/.conan2/profiles/default
[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu98
compiler.libcxx=libc++
compiler.version=14
os=Macos
```

**Comment by AbrilRBS** [comment]
Hi @simoncent thanks a lot for taking the time to report the issue, you're totally right, it's a byproduct of moving the detection api in the PR you pointed at. I've submitted #14837 and fixed the relevant tests that made this possible in the first time :)

The fix will be released in the next few days in the 2.0.13 version

**Comment by sizeak** [comment]
Excellent, thanks for the quick remediation.
